### PR TITLE
[auth] Add page for revoking access

### PR DESCRIPTION
**Pull request by gi-wg2**

### CODE REVIEW

File: docs/auth-revoke-consents.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Application requests a uninstall link.
+2. Application redirects the user to the given uri which opens a page hosted by WGTwo ID.  
+3. User confirms deleting the application by clicking a button in the page.
+4. User is redirected to a success page.
+
+In Step 2, the application may optionally include a `redirect_uri` and `state` query parameter.
+In that case, the user is redirected accordingly. 
+
+### Generate a uninstall link
```
Comment (jorunfa) [inline docs/auth-revoke-consents.md:24]:
```suggestion
### Generate an uninstall link
```

File: docs/auth-revoke-consents.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+title: Revoke consent
+topic: auth
+type: how-to
+---
+
+# Delete user
+
+We support a flow where you can initiate the user to remove access to your application.
+
+This should be used when a user choose to remove the application within your app.
```
Comment (jorunfa) [inline docs/auth-revoke-consents.md:11]:
```suggestion
We require app developers [support letting the user uninstall their app](why-support-user-app-uninstallation).
```

File: docs/auth-revoke-consents.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+# Delete user
+
+We support a flow where you can initiate the user to remove access to your application.
+
+This should be used when a user choose to remove the application within your app.  
+
+## Flow
+![](~/assets/images/auth-revoke-session.svg)
```
Comment (jorunfa) [inline docs/auth-revoke-consents.md:14]:
```suggestion
![Flowchart](~/assets/images/auth-revoke-session.svg)
```

File: docs/auth-revoke-consents.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Application requests a uninstall link.
+2. Application redirects the user to the given uri which opens a page hosted by WGTwo ID.  
+3. User confirms deleting the application by clicking a button in the page.
+4. User is redirected to a success page.
```
Comment (jorunfa) [inline docs/auth-revoke-consents.md:19]:
```suggestion
4. User is redirected back to the Application, and if valid, the Application must delete local user content (i.e. from a database) and wipe user sessions.
5. The user is shown a success page for uninstalling the app (or just app front-page).
```

File: docs/auth-revoke-consents.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+In Step 2, the application may optionally include a `redirect_uri` and `state` query parameter.
```
Comment (jorunfa) [inline docs/auth-revoke-consents.md:21]: I think this need to change; how does the Application wipe the user from the app database if the app doesn't provide a redirect uri? Also we should indicate that the `state` param is used to make sure the user really does want to uninstall/wipe out the user data, that no malicious actor has opened `https://voicebox.cc/wipe-this-user` in an iframe.

File: src/assets/images/auth-revoke-session.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
...bkJrZ0NvbG9yIjoid2hpdGUiLCJzZWN0aW9uQmtnQ29sb3IyIjoiI2ZmZjQwMCIsInRhc2tCb3JkZXJDb2xvciI6IiM1MzRmYmMiLCJ0YXNrQmtnQ29sb3IiOiIjOGE5MGRkIiwidGFza1RleHRMaWdodENvbG9yIjoid2hpdGUiLCJ0YXNrVGV4dENvbG9yIjoid2hpdGUiLCJ0YXNrVGV4dERhcmtDb2xvciI6ImJsYWNrIiwidGFza1RleHRPdXRzaWRlQ29sb3IiOiJibGFjayIsInRhc2tUZXh0Q2xpY2thYmxlQ29sb3IiOiIjMDAzMTYzIiwiYWN0aXZlVGFza0JvcmRlckNvbG9yIjoiIzUzNGZiYyIsImFjdGl2ZVRhc2tCa2dDb2xvciI6IiNiZmM3ZmYiLCJncmlkQ29sb3IiOiJsaWdodGdyZXkiLCJkb25lVGFza0JrZ0NvbG9yIjoibGlnaHRncmV5IiwiZG9uZVRhc2tCb3JkZXJDb2xvciI6ImdyZXkiLCJjcml0Qm9yZGVyQ29sb3IiOiIjZmY4ODg4IiwiY3JpdEJrZ0NvbG9yIjoicmVkIiwidG9kYXlMaW5lQ29sb3IiOiJyZWQiLCJsYWJlbENvbG9yIjoiYmxhY2siLCJlcnJvckJrZ0NvbG9yIjoiIzU1MjIyMiIsImVycm9yVGV4dENvbG9yIjoiIzU1MjIyMiIsImNsYXNzVGV4dCI6IiMxMzEzMDAiLCJmaWxsVHlwZTAiOiIjRUNFQ0ZGIiwiZmlsbFR5cGUxIjoiI2ZmZmZkZSIsImZpbGxUeXBlMiI6ImhzbCgzMDQsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlMyI6ImhzbCgxMjQsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSIsImZpbGxUeXBlNCI6ImhzbCgxNzYsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlNSI6ImhzbCgtNCwgMTAwJSwgOTMuNTI5NDExNzY0NyUpIiwiZmlsbFR5cGU2IjoiaHNsKDgsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlNyI6ImhzbCgxODgsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSJ9fSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)
```
Comment (jorunfa) [inline src/assets/images/auth-revoke-session.txt:3]: This link didn't work for me.

File: src/assets/images/auth-revoke-session.svg [hunk 7]
```diff
@@ -0,0 +1 @@
...ddle; font-size: 14px; font-weight: 400; font-family: Open-Sans, sans-serif;"><tspan x="75" dy="0">User</tspan></text></g><g><rect x="200" y="591" fill="#eaeaea" stroke="#666" width="150" height="65" rx="3" ry="3" class="actor"></rect><text x="275" y="623.5" dominant-baseline="central" alignment-baseline="central" class="actor" style="text-anchor: middle; font-size: 14px; font-weight: 400; font-family: Open-Sans, sans-serif;"><tspan x="275" dy="0">Browser</tspan></text></g><g><rect x="400" y="591" fill="#eaeaea" stroke="#666" width="150" height="65" rx="3" ry="3" class="actor"></rect><text x="475" y="623.5" dominant-baseline="central" alignment-baseline="central" class="actor" style="text-anchor: middle; font-size: 14px; font-weight: 400; font-family: Open-Sans, sans-serif;"><tspan x="475" dy="0">App server</tspan></text></g><g><rect x="600" y="591" fill="#eaeaea" stroke="#666" width="150" height="65" rx="3" ry="3" class="actor"></rect><text x="675" y="623.5" dominant-baseline="central" alignment-baseline="central" class="actor" style="text-anchor: middle; font-size: 14px; font-weight: 400; font-family: Open-Sans, sans-serif;"><tspan x="675" dy="0">WGTwo ID</tspan></text></g></svg>
```
Comment (jorunfa) [inline src/assets/images/auth-revoke-session.svg:1]: Maybe add arrow at end?
`App Server->>App Server: wipe user data`

File: docs/auth-revoke-consents.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+In that case, the user is redirected accordingly. 
+
+### Generate a uninstall link
+This endpoint requires authentication by a OAuth2 access token issues to your client using the `client_credentials` grant. 
+
+You also need to include a valid access token of the given user.
```
[the remark this answers is not in the document]
Comment (gi-wg2) [reply docs/auth-revoke-consents.md:27]: yep

**Pull request merged**